### ISSUE TITLE
  a bug in HttpClientBuilder.cs

### ISSUE THREAD

**Issue by liuyuedeyv** [ThreeMammals/Ocelot#441]
## Expected Behavior / New Feature
 if (httpClient != null)
            {
               **_client = httpClient;**
                return httpClient;
            }

## Actual Behavior / Motivation for New Feautre
Ocelot.Requester.HttpClientBuilder.Create  has one bug. when get a httpClient from cache ,not to the httpClient to the "_client",then when call "save" method,the parms "_client" is null.

## Steps to Reproduce the Problem

  1.
  1.
  1.

## Specifications

  - Version:
  - Platform:
  - Subsystem:

**Comment by TomPallister** [comment]
@liuyuedeyv thanks for your interest in the project!! 

What line is this?

All I see is line 41

```
 if (httpClient != null)
            {
                return httpClient;
            }
```

**Comment by TomPallister** [comment]
@liuyuedeyv ahh yes I see what you mean, sorry I was confused yesterday :) had a few beers!!! It is adding a null object to the cache! Thank you very much for spotting this. I will take a look!

**Comment by liuyuedeyv** [comment]
Thank very much! Good night!

**Comment by TomPallister** [comment]
@liuyuedeyv I will keep the issue open until it is released to NuGet 😄

**Comment by TomPallister** [comment]
released in nuget package 7.0.7 will be available on nuget soon!
